modules/wpml/wpml-config.php

```diff
@@ -565,6 +575,40 @@ private function get_field_attribute( SimpleXMLElement $field, $attribute_name )
 		return trim( (string) $attributes[ $attribute_name ] );
 	}
 
+	/**
+	 * Gets attributes values.
+	 *
+	 * @since 3.6
+	 *
+	 * @param  SimpleXMLElement $field A XML node.
+	 * @param  array            $attrs The attributes.
+	 * @param  bool             $first Tells if this is the first time in the function..
+	 * @return array|string
+	 */
+	private function get_field_attributes( SimpleXMLElement $field, array $attrs = array(), bool $first = true ) {
+		$name = $this->get_field_attribute( $field, 'name' );
+
+		$children = $field->children();
+
+		if ( $first && 0 === count( $children ) ) {
+			// No sub-attributes, don't go further and just return the attribute's value.
+			return $name;
+		}
+
+		if ( count( $children ) ) {
+			foreach ( $children as $child ) {
+				if ( ! isset( $attrs[ $name ] ) || ! is_array( $attrs[ $name ] ) ) {
+					$attrs[ $name ] = array();
+				}
+				$attrs[ $name ] = $this->get_field_attributes( $child, $attrs[ $name ], false );
+			}
+		} else {
+			$attrs[ $name ] = true;
+		}
+
+		return $attrs;
```

Comment:
```suggestion
                 if ( 1 === count( $attrs ) ) {
                     return reset( $attrs );
                 }
                 
		return $attrs;
```
Does it work? If so it will make `$first` parameter useless.
My point is to simplify the method and make it truly recursive.

Reply:
No, because it won't return the right values for the sub-attributes.

Reply:
My bad, this should work:
```php
/**
	 * Gets attributes values recursively.
	 *
	 * @since 3.6
	 *
	 * @param  SimpleXMLElement $field A XML node.
	 * @param  array            $attrs The attributes.
	 * @return array|string An array if there are sub-attributes, otherwise a string.
	 */
	private function get_field_attributes( SimpleXMLElement $field, array $attrs = array() ) {
		$name = $this->get_field_attribute( $field, 'name' );

		$children = $field->children();

		if ( 0 === count( $children ) ) {
			// No sub-attributes, don't go further and just return the attribute's value.
			return $name;
		}

		if ( count( $children ) ) {
			foreach ( $children as $child ) {
				if ( ! isset( $attrs[ $name ] ) || ! is_array( $attrs[ $name ] ) ) {
					$attrs[ $name ] = array();
				}

				$sub = $this->get_field_attributes( $child, $attrs[ $name ] );

				if ( is_string( $sub ) ) {
					$sub = array( $sub => true );
				}

				$attrs[ $name ] = array_merge( $attrs[ $name ], $sub );
			}
		} else {
			$attrs[ $name ] = true;
		}

		return $attrs;
	}
```
What do you think?